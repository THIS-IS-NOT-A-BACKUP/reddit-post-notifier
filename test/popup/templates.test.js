import { JSDOM } from 'jsdom';
import getTemplates from '../../scripts/popup/templates';
import html from '../../extension/popup.html';

beforeAll(() => {
    window.document.body.innerHTML = new JSDOM(html).window.document.body.innerHTML;
});

describe('templates', () => {
    test('should match snapshot', () => {
        expect(getTemplates()).toMatchSnapshot();
    });
});
