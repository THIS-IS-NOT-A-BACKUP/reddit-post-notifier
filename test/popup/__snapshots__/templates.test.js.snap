// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates should match snapshot 1`] = `
Object {
  "noPosts": <DocumentFragment>
    
      
    <div
      class="no-posts"
    >
      No unread posts
    </div>
    
    
  </DocumentFragment>,
  "postList": <DocumentFragment>
    
      
    <ul
      class="post-list"
    />
    
    
  </DocumentFragment>,
  "postListRow": <DocumentFragment>
    
      
    <li
      tabindex="1"
    >
      
        
      <button
        class="check-mark"
      >
        
          
        <svg
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          
            
          <use
            href="#check-mark-path"
          >
            
          
          </use>
        </svg>
        
        
      </button>
      
        
      <a
        href="#"
      />
      
      
    </li>
    
    
  </DocumentFragment>,
  "queryList": <DocumentFragment>
    
      
    <ul
      class="query-list"
    >
      
      
    </ul>
    
    
  </DocumentFragment>,
  "queryListRow": <DocumentFragment>
    
      
    <li
      tabindex="1"
    >
      
        
      <button
        class="check-mark"
      >
        
          
        <svg
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          
            
          <use
            href="#check-mark-path"
          >
            
          
          </use>
        </svg>
        
        
      </button>
      
        
      <div
        class="query-name"
      />
      
        
      <button
        class="arrow-right"
      >
        <img
          src="arrowhead-right.svg"
        />
      </button>
      
      
    </li>
    
    
  </DocumentFragment>,
}
`;
